pkg/pipeline/service.go

```diff
@@ -193,6 +193,36 @@ func (s *Service) Update(ctx context.Context, pipelineID string, cfg Config) (*I
 	return pl, err
 }
 
+// UpdateDLQ will update a pipeline DLQ config.
+func (s *Service) UpdateDLQ(ctx context.Context, pipelineID string, cfg DLQ) (*Instance, error) {
+	pl, err := s.Get(ctx, pipelineID)
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.Plugin == "" {
+		return nil, cerrors.New("DLQ plugin must be provided")
+	}
+	if cfg.WindowSize < 0 {
+		return nil, cerrors.New("DLQ window size must be non-negative")
```

Comment:
what does it mean if window size is 0?

Reply:
Window size 0 means that no records are monitored, all nacked records are rerouted to the DLQ without a limit so the pipeline doesn't stop.